client/src/components/Reports/index.js

```diff
@@ -0,0 +1,358 @@
+  ) => {
+    getData(suggestionValue).then(data => {
+      this.setState({ data, ticker: suggestionValue });
+    });
+  };
+
+  // TODO: Figure out how to make the first tab be selected by default
+  handleTabChange = (event, value) => {
+    this.setState({ value });
+  };
+
+  // TODO: Move these helper functions to a separate file for use here and in the stock ticker
+  changePercent = (close, start) => {
+    // function for calculating the change of a stocks gain/loss by %
+    let deduct = close - start;
+    let divide = deduct / start;
+    let solution = divide * 100;
+    if (solution > 0) {
+      return "+" + solution.toFixed(2);
+    }
+    return solution.toFixed(2);
+  };
+
+  changePoints = (close, start) => {
+    // calculates the change of a stocks gain/loss by points
+    let solution = close - start;
+    if (solution > 0) {
+      return "+" + solution.toFixed(1);
+    }
+    return solution.toFixed(1);
+  };
+
+  decimalToFixed = input => {
+    // truncates the numbers following the decimal to two digits
+    input = parseFloat(input).toFixed(2);
+    return input;
+  };
+
+  shortenVolume = num => {
+    // Crunches the length of the volume into a smaller number while inserting a decimal point and character representing the amount
+    let str,
+      suffix = "";
+
+    let decimalPlaces = 2 || 0;
+
+    num = +num;
+
+    let factor = Math.pow(10, decimalPlaces);
+
+    if (num < 1000) {
+      str = num;
+    } else if (num < 1000000) {
+      str = Math.floor(num / (1000 / factor)) / factor;
+      suffix = "K";
+    } else if (num < 1000000000) {
+      str = Math.floor(num / (1000000 / factor)) / factor;
+      suffix = "M";
+    } else if (num < 1000000000000) {
+      str = Math.floor(num / (1000000000 / factor)) / factor;
+      suffix = "B";
+    }
+    return str + suffix;
+  };
+
+  render() {
+    const { classes } = this.props;
+    const { value, data } = this.state;
+
+    const autosuggestProps = {
+      renderInputComponent,
+      suggestions: this.state.suggestions,
+      onSuggestionsFetchRequested: this.handleSuggestionsFetchRequested,
+      onSuggestionsClearRequested: this.handleSuggestionsClearRequested,
+      onSuggestionSelected: this.onSuggestionSelected,
+      getSuggestionValue,
+      renderSuggestion
+    };
+
+    return (
+      <React.Fragment>
+        <CssBaseline />
+        <div className={classes.root}>
+          <Grid container justify="center">
+            <Grid
+              spacing={24}
+              alignItems="center"
+              justify="center"
+              container
+              className={classes.grid}
+            >
+              <Grid item xs={12}>
+                <div className={classes.topBar}>
+                  <div className={classes.block}>
+                    <Typography variant="h2">{this.state.ticker}</Typography>
+                  </div>
+                  <div>
+                    <Autosuggest
```

Comment:
This autosuggest is really nice. The setup is really intresting where did you find this information on doing the autosuggest?